base_layer/wallet/src/util/encryption.rs

```diff
@@ -43,8 +43,10 @@ pub trait Encryptable<C> {
     const CLIENT_KEY_VALUE: &'static [u8] = b"CLIENT_KEY_VALUE";
 
     fn domain(&self, field_name: &'static str) -> Vec<u8>;
-    fn encrypt(&mut self, cipher: &C) -> Result<(), String>;
-    fn decrypt(&mut self, cipher: &C) -> Result<(), String>;
+    fn encrypt(self, cipher: &C) -> Result<Self, String>
```

Comment:
```suggestion
    fn encrypt(mut self, cipher: &C) -> Result<Self, String>
```
This leads to bad behavior as marked out in the other functions. 
Self is dropped after the function, and because its not mut, the values cannot be changed or zeroized. Thus after the function when self is dropped, the sensitive data is still in memory.

Reply:
I refactored the instances of these methods to receive as input `mut self`, zeroize its sensitive contents and drop the values, after function returning. See comment above